infra/datalake/processing_stack.py

```diff
@@ -0,0 +1,309 @@
+                    f"{task}-function",
+                    handler=f"processing.{task}.task.lambda_handler",
+                    runtime=aws_lambda.Runtime.PYTHON_3_6,
+                    code=aws_lambda.Code.from_asset(
+                        path="..",
+                        bundling=core.BundlingOptions(
+                            # pylint:disable=no-member
+                            image=aws_lambda.Runtime.PYTHON_3_6.bundling_docker_image,
+                            command=["backend/processing/bundle.bash", f"{task}"],
+                        ),
+                    ),
+                )
+
+                step_tasks[task] = aws_stepfunctions_tasks.LambdaInvoke(
+                    self,
+                    f"{task}",
+                    lambda_function=lambda_function,
+                    input_path=creation_tasks[task].get("input_path", "$"),
+                    output_path=creation_tasks[task].get("output_path", "$"),
+                    result_path=creation_tasks[task].get("result_path", "$"),
+                    payload_response_only=True,
+                )
+
+                Tags.of(lambda_function).add("ApplicationLayer", "data-processing")
+
+            # aws batch jobs
+            if creation_tasks[task]["type"] == "batch":
+
+                job_definition = aws_batch.JobDefinition(
+                    self,
+                    f"{task}-job",
+                    container=aws_batch.JobDefinitionContainer(
+                        image=aws_ecs.ContainerImage.from_asset(
+                            directory=f"../backend/processing/{task}",
+                        ),
+                        memory_limit_mib=3900 if ENV == "prod" else 500,
+                        vcpus=1,
+                    ),
+                    retry_attempts=4,
+                )
+
+                job_command = [
+                    "--dataset-id",
+                    "Ref::dataset_id",
+                    "--version-id",
+                    "Ref::version_id",
+                    "--type",
+                    "Ref::type",
+                    "--metadata-url",
+                    "Ref::metadata_url",
+                    "--dataset-id",
+                ]
+                job_environment = {"BATCH_JOB_FIRST_ITEM_INDEX": "Ref::first_item"}
+
+                job_payload_data = {
+                    "dataset_id.$": "$.dataset_id",
+                    "version_id.$": "$.version_id",
+                    "type.$": "$.type",
+                    "metadata_url.$": "$.metadata_url",
+                }
+                job_payload_data_parallel = {"first_item.$": "$.content.first_item"}
+                job_payload_single = aws_stepfunctions.TaskInput.from_object(job_payload_data)
+                job_payload_parallel = aws_stepfunctions.TaskInput.from_object(
+                    {**job_payload_data, **job_payload_data_parallel}
+                )
+
+                if creation_tasks[task]["parallel"]:
```

Comment:
`creation_tasks[task]` is used so many times, we should probably rename `task` to `task_name` and pull out `task = creation_tasks[task_name]`.

Reply:
Done in c16a3329472d71183fdbf0dbaa4928125a8ac93a